shell/common/rasterizer.h

```diff
@@ -20,78 +20,367 @@
+  ///             apply to snapshot delegates as well.
+  ///
+  /// @return     The weak pointer to the snapshot delegate.
+  ///
   fml::WeakPtr<SnapshotDelegate> GetSnapshotDelegate() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Sometimes, it may be necessary to render the same frame again
+  ///             without having to wait for the framework to build a whole new
+  ///             layer tree describing the same contents. One such case is when
+  ///             external textures (video or camera streams for example) are
+  ///             updated in an otherwise static layer tree. To support this use
+  ///             case, the rasterizer holds onto the last rendered layer tree.
+  ///
+  /// @return     A pointer to the last layer or `nullptr` if this rasterizer
+  ///             has never rendered a frame.
+  ///
   flutter::LayerTree* GetLastLayerTree();
 
+  //----------------------------------------------------------------------------
+  /// @brief      Draws a last layer tree to the render surface. This may seem
+  ///             entirely redundant at first glance. After all, on surface loss
+  ///             and re-acquisition, the framework generates a new layer tree.
+  ///             Otherwise, why render the same contents to the screen again?
+  ///             This is used as an optimization in cases where there are
+  ///             external textures (video or camera streams for example) in
+  ///             referenced in the layer tree. These textures may be updated at
+  ///             a cadence different from that of the the Flutter application.
+  ///             Flutter can re-render the layer tree with just the updated
+  ///             textures instead of waiting for the framework to do the work
+  ///             to generate the layer tree describing the same contents.
+  ///
   void DrawLastLayerTree();
 
+  //----------------------------------------------------------------------------
+  /// @brief      Gets the registry of external textures currently in use by the
+  ///             rasterizer. These textures may be updated at a cadence
+  ///             different from that of the Flutter application. When an
+  ///             external texture is referenced in the Flutter layer tree, that
+  ///             texture is composited within the Flutter layer tree.
+  ///
+  /// @return     A pointer to the external texture registry.
+  ///
   flutter::TextureRegistry* GetTextureRegistry();
 
+  //----------------------------------------------------------------------------
+  /// @brief      Takes the next item from the layer tree pipeline and executes
+  ///             the GPU thread frame workload for that pipeline item to render
+  ///             a frame on the on-screen surface.
+  ///
+  ///             Why does the draw call take a layer tree pipeline and not the
+  ///             layer tree directly?
```

Comment:
Maybe add a note like "|Rasterizer::DoDraw| takes a layer tree and rasterizes it; The |Rasterizer::Draw| takes the pipeline as it needs to do additional pipeline maintenance work".

Reply:
Done.